src/praisonai-mcp/praisonai_mcp/mcp_server/adapters/memory.py

```
@@ -14,91 +20,75 @@
-            else:
-                data = memory.get_all()
+            data = memory.get_all_memories()
+            if user_id:
+                data = [
+                    m for m in data
+                    if str((m.get("metadata") or {}).get("user_id")) == str(user_id)
+                ]
             return str(data)
         except ImportError:
             return "Error: Memory module not available"
         except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)